### PR TITLE
Add detect_numa script

### PR DESCRIPTION
**Pull request by kdvalin**
This PR introduces `detect_numa` to fetch NUMA information, allowing scripts to get NUMA information in the same way.

This also adds tests for `detect_numa`, using a generic workflow, allowing easy expansion later on.

### CODE REVIEW

**Comment by dvalinrh** [comment]
LGTM

**Comment by kdvalin** [comment]
@dvalinrh Updated with some new codepaths, review please

File: detect_numa [hunk 1]
```diff
@@ -26,11 +26,21 @@ usage()
...
 }
 
+# Verifies that NUMA exists on the system
+has_numa()
+{
+    local numa_nodes=`lscpu | grep NUMA | grep CPU | wc -l`
+    if [ -z "$numa_nodes" ] || [ $numa_nodes -le 0 ]; then
+        echo 0
+    fi
+    echo 1
```
Comment (dvalinrh) [inline detect_numa:36]: you should have an else here, we will be echoing both a 0 and a 1.

**Review by dvalinrh — CHANGES_REQUESTED** [review]

File: detect_numa (hunk 1 above)
Comment (kdvalin) [reply detect_numa]: Fixed

**Comment by dvalinrh** [comment]
LGTM.

**Pull request merged**